src/modelinstance.cpp

```diff
@@ -104,99 +117,213 @@ Status ModelInstance::loadInputTensors(const ModelConfig& config, const DynamicM
+        }
+    }
+    for (const auto& [name, _] : config.getLayouts()) {
+        if (hasInputWithName(network, name) && config.getMappingInputByKey(name) != "") {
+            SPDLOG_WARN("Config layout - {} is mapped by {}. Changes will not apply", name, config.getMappingInputByKey(name));
+            return StatusCode::CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME;
+        } else if (hasOutputWithName(network, name) && config.getMappingOutputByKey(name) != "") {
+            SPDLOG_WARN("Config layout - {} is mapped by {}. Changes will not apply", name, config.getMappingOutputByKey(name));
+            return StatusCode::CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME;
+        } else if (!hasInputWithName(network, name) && !hasOutputWithName(network, name) && !hasInputWithName(network, config.getRealInputNameByValue(name)) && !hasOutputWithName(network, config.getRealOutputNameByValue(name))) {
+            SPDLOG_WARN("Config layout - {} not found in network", name);
+            return StatusCode::CONFIG_LAYOUT_IS_NOT_IN_NETWORK;
+        }
+    }
+    return StatusCode::OK;
+}
 
-        SPDLOG_DEBUG("Network shape for input: {} - {}; final shape {}", name,
-            TensorInfo::shapeToString(networkShapes[name]),
-            TensorInfo::shapeToString(shape));
+InferenceEngine::Layout getTensorLayout(const ModelConfig& config, const std::string& name) {
+    InferenceEngine::Layout layout = InferenceEngine::Layout::ANY;
+    if (!config.getLayout().empty()) {
+        layout = TensorInfo::getLayoutFromString(config.getLayout());
+    } else if (config.getLayouts().size() > 0) {
+        auto mappedName = config.getMappingInputByKey(name);
+        auto it = config.getLayouts().find(mappedName == "" ? name : mappedName);
+        if (it != config.getLayouts().end()) {
+            layout = TensorInfo::getLayoutFromString(it->second);
+        }
+    }
+    return layout;
+}
 
-        if (networkShapes[name] != shape) {
-            reshapeRequired = true;
-            networkShapes[name] = shape;
+Status addPrePostProcessingSteps(const ModelConfig& config, std::shared_ptr<ov::Function>& network, const std::string& modelName, model_version_t modelVersion) {
+    if (!config.getLayout().empty()) {
+        ov::preprocess::PrePostProcessor preproc(network);
+        std::string tensorLayout = config.getLayout();
+        std::string networkLayout = tensorLayout == "NCHW" ? "NHWC" : "NCHW";
+        SPDLOG_INFO("model: {}, version: {}; Adding preprocessing step: {}; Tensor Layout:{}; Network Layout:{}", modelName, modelVersion, config.getLayout(), tensorLayout, networkLayout);
+        preproc.input().tensor().set_layout(ov::Layout(tensorLayout));
+        preproc.input().network().set_layout(ov::Layout(networkLayout));
+        try {
+            network = preproc.build();
+        } catch (std::exception& e) {
+            SPDLOG_ERROR("Cannot change single input layout to {}", tensorLayout);
```

Comment:
logger,

Reply:
+ in other places as well

Reply:
Adding logger would be misleading I think. This can be triggered in predict path when the model is shape=auto, when model manager watcher cycle detected change to config.json and also in REST path with Model Control API.